syntax = "proto3";

package engine;

message ColorParams {
    int32 r = 1;
    int32 g = 2;
    int32 b = 3;
    int32 a = 4;
}

message AtlasParams {
    string texture                  = 1;
    int32 tile_size                 = 2;
    int32 gutter                    = 3;
    map<string, int32> tile_names   = 4;
}

message EngineParams {
    string atlas                = 1;
    int32 width                 = 2;
    int32 height                = 3;
    int32 render_scale          = 4;
    ColorParams clear_color     = 5;
}

message DunGenParams {
    int32 room_count    = 1;
    int32 min_room_size = 2;
    int32 max_room_size = 3;
    int32 map_width     = 4;
    int32 map_height    = 5;
    int32 expand_chance = 6; // Expressed as a percentage 0-100
}
